spring-context/src/main/java/org/springframework/scheduling/annotation/ScheduledAnnotationBeanPostProcessor.java

```diff
@@ -398,7 +397,7 @@ protected void processScheduled(Scheduled scheduled, Method method, Object bean)
 			Set<ScheduledTask> tasks = new LinkedHashSet<>(4);
 
 			// Determine initial delay
-			long initialDelay = scheduled.initialDelay();
+			long initialDelay = scheduled.initialDelayTimeUnit().toMillis(1) * scheduled.initialDelay();
```

Comment:
Wouldn't it be more idiomatic to use `TimeUnit.MILLISECONDS.convert(scheduled.initialDelay(), scheduled.initialDelayTimeUnit())`?

Reply:
Seems better, fixed !